internal/server/accounts.go

```
@@ -43,6 +44,16 @@ func (s *Store) CreateAccount(email string, kdf crypto.KdfParams, publicKey []by
 	if len(encPublicKey) > 0 {
 		encPub, encSig = encPublicKey, encKeySig
 	}
+	// The UNIQUE(email) index is binary, so a row written before emails were
+	// normalized would not block a lower-cased twin of the same mailbox; check
+	// case-insensitively first. Concurrent signups of the same normalized email
+	// still serialize on the unique index below.
+	var exists int
+	if err := s.db.QueryRow(`SELECT 1 FROM accounts WHERE email = ? COLLATE NOCASE`, email).Scan(&exists); err == nil {
```

golangci-lint (2.12.2)

[error] (*database/sql.DB).QueryRow must not be called. use (*database/sql.DB).QueryRowContext

(noctx)